src/main/java/com/opendxl/streaming/client/Channel.java

```diff
@@ -1246,6 +1247,124 @@ private String logConsumerId() {
         return "consumer " + consumerId;
     }
 
+    /**
+     * Update consumer filter with updated filter values
+     *
+     * @param topics Topic list.
+     * @throws ConsumerError  if the consumer associated with the channel does not
+     *                        exist on the server.
+     * @throws PermanentError if no consumer instance-id was specified.
+     * @throws TemporaryError if the subscription attempt fails.
+     */
+    public void updateFilter(final Map<String, Map<String, Object>> filter)
+            throws PermanentError, TemporaryError, ClientError {
+        if (consumerId == null || consumerId.isEmpty()) {
+            String error = "Consumer instance id can not be null or empty";
+            logger.error(error);
+            throw new PermanentError(error);
+        }
+
+        Gson gson = new Gson();
+        byte[] body = null;
+        if (null == filter || filter.isEmpty()) {
+            Topics topicsToBeSubscribed = new Topics(null);
+            body = gson.toJson(topicsToBeSubscribed).getBytes();
+        } else {
+            UpdateFilterRequest payload = new UpdateFilterRequest(filter);
+            body = gson.toJson(payload).getBytes();
+        }
+
+        StringBuilder api = new StringBuilder(consumerPathPrefix)
+                .append("/consumers/")
+                .append(consumerId)
+                .append("/updateFilter");
+
+        final String logMessage = new StringBuilder(logConsumerId())
+                .append(" to ").toString();
+        try {
+
+            request.post(api.toString(), body, UPDATE_FILTER_ERROR_MAP);
+            if (logger.isDebugEnabled()) {
+                logger.debug("Update filter call " + logMessage);
+            }
+
+        } catch (final ClientError error) {
+            error.setApi("update-filter");
+            logger.error("Failed to update filter " + logMessage, error);
+            throw error;
+        }
+
+    }
+
+    public String callConsumerGroupDescribe(final String consumerGroupName)
+            throws PermanentError, TemporaryError {
+
+        acquireAndEnsureChannelIsActive();
```

Comment:
Do we need this -   acquireAndEnsureChannelIsActive() ?

Reply:
In this case it is required as we are creating a new Channel for consumer-lag api if we don't have active channel this method will handle that.